File: topics/control-flow/exercises/rect_move.c
#define X_SPEED 5

void ????(int width, int height)
{
    float x = 0, y;
    
    y = (screen_height() - height) / 2;
    
    while ( (x + width) < screen_width() )
    {
        clear_screen();
        
        fill_rectangle(ColorRed, x, y, width, height);
        x = x + X_SPEED;
        
        refresh_screen();
    }
}

File: topics/control-flow/exercises/RectMove.pas
const X_SPEED = 5;

procedure ????(width, height: Integer);
var
    y: Single;
    x: Single = 0;
begin
  y := (ScreenHeight() - height) / 2;
  
  while (x + width) < ScreenWidth() do
  begin
    ClearScreen();
    
    FillRectangle(ColorRed, x, y, width, height);
    x := x + X_SPEED;
    
    RefreshScreen();
  end; 
end;